### ISSUE TITLE
Multiline comments broken if contain code with `###`

### ISSUE THREAD

**Issue by Cpt0r** [jashkenas/coffeescript#4897]
```
> cat test.coffee
###
console.log '### this should be ignored, not compiled'

console.log '### this too'

console.log '####### and this'
###

> bin/coffee -p test.coffee
(function() {
  /*
  console.log '*/
  this(should(be(ignored, !compiled`console.log `)));

  /* this too'
//##

console.log '*/

}).call(this);
```

Using latest master: 571e9df335c51caa4c363ba1662998acb332c1e4

**Comment by Inve1951** [comment]
i was under the impression that coffee took care of this by requiring the opening and closing tag to share the same indentation level. i remember getting indentation errors when commenting out code parts without matching them in my early days with coffee.

**Comment by GeoffreyBooth** [comment]
@Inve1951 Comments were refactored between 1 and 2, to allow block (`###`) comments to be inserted anywhere, and to allow line (`#`) comments to be output into JS. This wasn’t considered a breaking change, because all previously allowed block comments were still allowed in 2; and _more_ cases of block comments became allowed.

However I don’t seem to be able to replicate the “same indentation” requirement you describe. At least in 1.12.7, open and closing `###` delimiters don’t seem to need to be at the same level: http://coffeescript.org/v1/#try:%23%23%23%0A%091%0A%09%09%23%23%23. Maybe that was the case in a much older version of CoffeeScript.

**Comment by Inve1951** [comment]
[v2](http://coffeescript.org/v2/#try:a%20%3D%20-%3E%0A%23%23%23%0A%091%0A%23%23%23%0A%20%20b) vs [v1](http://coffeescript.org/v1/#try:a%20%3D%20-%3E%0A%23%23%23%0A%091%0A%23%23%23%0A%20%20b)
not quite what i described earlier. in v1 the comment opening tag has to be on current indent level or it gives an indent error.

**Comment by GeoffreyBooth** [comment]
Yeah, that’s to be expected. In v1 comments could only exist where expressions could, like where you could put a parenthesized block, whereas in v2 they can now exist anywhere.

**Comment by Inve1951** [comment]
OP's issue is quite the edge case, indeed. however i feel like there could be a clean solution to it. for example escaping the first hashtag with a backslash.
current workaround is to escape every other `#` like so:
```coffee
###
console.log '\##\# this should be ignored, not compiled'

console.log '\##\# this too'

console.log '\##\##\##\# and this'
###
```

**Comment by GeoffreyBooth** [comment]
I think the most reasonable solution is to just use an editor that allows commenting out long selections of text. Almost any editor I’ve used, whether a simple text editor or an IDE, allows you to select multiple lines of text and press <kbd>Cmd</kbd>-<kbd>/</kbd> or <kbd>Ctrl</kbd>-<kbd>/</kbd> to insert `#`s at the beginning of the line.

**Comment by Inve1951** [comment]
neat, learned something new
